gateway/src/main/java/org/georchestra/gateway/security/oauth2/OAuth2Configuration.java

```diff
@@ -105,11 +106,21 @@ public ReactiveJwtDecoderFactory<ClientRegistration> idTokenDecoderFactory(
         return (clientRegistration) -> {
             ProviderDetails providerDetails = clientRegistration.getProviderDetails();
             String jwkSetUri = providerDetails.getJwkSetUri();
-            return NimbusReactiveJwtDecoder//
-                    .withJwkSetUri(jwkSetUri)//
-                    .jwsAlgorithm(SignatureAlgorithm.RS256)//
-                    .webClient(oauth2WebClient)//
-                    .build();
+            if ((jwkSetUri != null) && (!jwkSetUri.isEmpty())) {
+                return NimbusReactiveJwtDecoder.withJwkSetUri(jwkSetUri).webClient(oauth2WebClient).build();
+            } else {
+                ReactiveOidcIdTokenDecoderFactory idTokenDecoderFactory = new ReactiveOidcIdTokenDecoderFactory();
```

Comment:
Just to confirm : Google login still works after this changes, I remember this was made to make FranceConnect authentication works

Reply:
Google has a jwkSetUri so it uses `NimbusReactiveJwtDecoder`, and FranceConnect does not have a jwkSetUri and uses `ReactiveOidcIdTokenDecoderFactory` (which also uses internally `NimbusReactiveJwtDecoder` by doing something similar to the commented code below).